File: static/code/sample/tethered/gdl-seclipsen-a.cs
// In this sample:
// Touch on pin 2, status LED turns on
// Touch on pin 7, status LED turns off


using System.Net.NetworkInformation;
using GHIElectronics.DUELink;

var availablePort = DUELinkController.GetConnectionPort();
var duelink = new DUELinkController(availablePort);

void SetStatLed(bool value) {
    var high = value ? 1 : 0;
    var low = !value ? 1 : 0;
    duelink.Engine.ExecuteCommand($"Statled({high},{low},0)");
}

bool Touched(int pin) {
   var ret = duelink.Engine.ExecuteCommand($"PulseIn({pin},1000, 0, 100)");

    if (ret > 110) {
        return true;
    }

    return false;
   
}

while (true) {
    if (Touched(2))
        SetStatLed(true);
    if (Touched(7))
        SetStatLed(false);

    Thread.Sleep(50);
}

File: static/code/sample/tethered/gdl-seclipsen-a.py
# In this sample:
# Touch on pin 2, status LED turns on
# Touch on pin 7, status LED turns off

import time
from DUELink.DUELinkController import DUELinkController

availablePort = DUELinkController.GetConnectionPort()
duelink = DUELinkController(availablePort)

# Methods
def SetStatLed(value: bool):
    high = 1 if value else 0
    low = 0 if value else 1
    duelink.Engine.ExecuteCommand(f"Statled({high},{low},0)")

def Touched(pin: int) -> bool:
    ret = duelink.Engine.ExecuteCommand(f"PulseIn({pin},1000, 0, 100)")
    return ret > 110

while True:
    if Touched(2):
        SetStatLed(True)
    if Touched(7):
        SetStatLed(False)

    time.sleep(0.05)  # 50 ms delay